cartographer_grpc/framework/service.h

```diff
@@ -32,16 +32,21 @@ class Service : public ::grpc::Service {
  public:
   Service(const std::string& service_name,
           const std::map<std::string, RpcHandlerInfo>& rpc_handlers);
-  void StartServing(
-      const std::vector<::grpc::ServerCompletionQueue*>& completion_queues);
+  void StartServing(std::vector<CompletionQueueThread>& completion_queues);
+  void HandleEvent(Rpc::State state, Rpc* rpc, bool ok);
+  void StopServing();
 
  private:
   void RequestNextMethodInvocation(
       int method_index, Rpc* rpc,
       ::grpc::ServerCompletionQueue* completion_queue);
 
+  void HandleNewConnection(Rpc* rpc, bool ok);
```

Comment:
I followed gRPC in the naming. The function to pull events from t.e queue is declared as  `bool ServerCompletionQueue::Next(void** tag, bool* ok)`. I kind of like `ok` because it's short. We could also use `success` I guess if you prefer that.